### PR TITLE
fix: throw error when conflicting connection settings for same instance

### PR DESCRIPTION
**Pull request by shubha-rajan**
depends on #83

### CODE REVIEW

File: test/connector.ts [hunk 1]
```diff
@@ -221,6 +221,9 @@ t.test('start only a single instance info per connection name', async t => {
             throw new Error('should only initialize once');
           }
           hasInstance = true;
+          return {
+            connectionType: 'PUBLIC',
```
Comment (ruyadorno) [inline test/connector.ts:225]:
```suggestion
            ipType: 'PUBLIC',
```

File: test/tap-testdir-sqladmin-fetcher-getEphemeralCertificate-no-certificate/credentials
Comment (ruyadorno) [inline test/tap-testdir-sqladmin-fetcher-getEphemeralCertificate-no-certificate/credentials:1]: why are these files popping up here? maybe they were part of the larger PR?

Comment (shubha-rajan) [reply test/tap-testdir-sqladmin-fetcher-getEphemeralCertificate-no-certificate/credentials:1]: oops, accidentally committed them. they're generated files, I believe.

Comment (enocom) [reply test/tap-testdir-sqladmin-fetcher-getEphemeralCertificate-no-certificate/credentials:1]: Should we add these to the gitignore?

**Review by ruyadorno — APPROVED** [review]

**Pull request merged**